shell/mixins/browser-tab-visibility.js

```diff
@@ -0,0 +1,24 @@
+import { NORMAN } from '@shell/config/types';
+
+export default {
+  methods: {
+    setTabVisibilityListener(isAdd) {
+      const method = isAdd ? 'addEventListener' : 'removeEventListener';
```

Comment:
This needs to be exit early to cover cases where there is no rancher BE (harvester, desktop, etc). Think we could probably start with something based on `this.$config.rancherEnv` or a new value in `isSingleProduct` (and applied to the harvester and epinio types). Probably best to bring up in chat

Reply:
Taken in account on this new commit

Reply:
Rancher Desktop only makes use of components. We do embed Rancher Dashboard, but I don't anticipate any changes to layouts being a concern.

Reply:
@rak-phillip Ah, for that embedded dashboard case, does it support norman/v3 requests? This PR does make ping out to `rancher/findAll` when the user returns to the browser's tab

Reply:
Only Steve us supported at this point..

Reply:
Ok, from my understanding then this PR will break your use case. So we should also exclude with something like `this.$config.rancherEnv === 'desktop'`

@aalves08 ^